src/bootstrap/src/core/build_steps/run.rs

```diff
@@ -37,7 +37,14 @@ impl Step for BuildManifest {
     fn run(self, builder: &Builder<'_>) {
         // This gets called by `promote-release`
         // (https://github.com/rust-lang/promote-release).
-        let mut cmd = builder.tool_cmd(Tool::BuildManifest);
+        let mut cmd = command(
+            builder
+                .ensure(tool::BuildManifest {
```

Comment:
Could you please add `tool::BuildManifest::new(target)`, so that the tool correctly auto-selects the stage1 compiler, and we don't pass a different build compiler by accident? I know that there's also an assert in `BuildManifest`'s implementation, but this would be clearer and harder to make a mistake.